console/v1/types_console_link.go

```diff
@@ -47,7 +47,14 @@ type ApplicationMenuSpec struct {
 // NamespaceDashboardSpec is a specification of namespaces in which the dashboard link should appear.
 type NamespaceDashboardSpec struct {
 	// namespaces is an array of namespace names in which the dashboard link should appear.
-	Namespaces []string `json:"namespaces"`
+	Namespaces []string `json:"namespaces,omitempty"`
```

Comment:
Adding `omitempty` here since if `Location` is specified to `NamespaceDashboard` and should apply to all the Namespaces then the `NamespaceDashboardSpec` should be omitted.
Adding `+optional` due to that.